{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('/Users/estorrs/Downloads/13059_2014_3364_MOESM1_ESM/supplementary_table_2-Table 1.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[['CHR', 'POS', 'VariationName', 'VT', 'VLENGTH', 'Ancestralallele', 'Consequencespecificallele']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>VariationName</th>\n",
       "      <th>VT</th>\n",
       "      <th>VLENGTH</th>\n",
       "      <th>Ancestralallele</th>\n",
       "      <th>Consequencespecificallele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>13</td>\n",
       "      <td>106835388</td>\n",
       "      <td>rs9519977</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>116935068</td>\n",
       "      <td>rs10924081</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>146978905</td>\n",
       "      <td>rs4059271</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>3</td>\n",
       "      <td>na</td>\n",
       "      <td>T/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>140933740</td>\n",
       "      <td>rs55726039</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1</td>\n",
       "      <td>T</td>\n",
       "      <td>T/C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>5775231</td>\n",
       "      <td>rs1811510</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1</td>\n",
       "      <td>T</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10</td>\n",
       "      <td>68591682</td>\n",
       "      <td>rs10762111</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>T/A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>22</td>\n",
       "      <td>50704357</td>\n",
       "      <td>rs11704249</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>235946882</td>\n",
       "      <td>na</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>3</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>91200757</td>\n",
       "      <td>rs76306981</td>\n",
       "      <td>INDEL</td>\n",
       "      <td>2</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>111924518</td>\n",
       "      <td>rs11814770</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1</td>\n",
       "      <td>T</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR        POS VariationName     VT  VLENGTH Ancestralallele  \\\n",
       "0   13  106835388     rs9519977    SNP        1               A   \n",
       "1    1  116935068    rs10924081    SNP        1               G   \n",
       "2    3  146978905     rs4059271  INDEL        3              na   \n",
       "3    8  140933740    rs55726039    SNP        1               T   \n",
       "4    3    5775231     rs1811510    SNP        1               T   \n",
       "5   10   68591682    rs10762111    SNP        1               A   \n",
       "6   22   50704357    rs11704249    SNP        1               C   \n",
       "7    2  235946882            na  INDEL        3              na   \n",
       "8    1   91200757    rs76306981  INDEL        2              na   \n",
       "9   10  111924518    rs11814770    SNP        1               T   \n",
       "\n",
       "  Consequencespecificallele  \n",
       "0                        na  \n",
       "1                       A/G  \n",
       "2                       T/G  \n",
       "3                       T/C  \n",
       "4                        na  \n",
       "5                       T/A  \n",
       "6                       C/T  \n",
       "7                        na  \n",
       "8                        na  \n",
       "9                        na  "
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_alternate(reference, consequence):\n",
    "    pieces = consequence.split('/')\n",
    "    if pieces[0] == reference:\n",
    "        return pieces[1]\n",
    "    return pieces[0]\n",
    "\n",
    "def populate_alternate(df):\n",
    "    filtered = pd.DataFrame()\n",
    "    for i, row in df.iterrows():\n",
    "        # just do snps for now\n",
    "        if row['VLENGTH'] == 1:\n",
    "            if row['Ancestralallele'] != 'na' and row['Consequencespecificallele'] != 'na':\n",
    "                row['Alternate'] = get_alternate(row['Ancestralallele'], row['Consequencespecificallele'])\n",
    "                filtered = filtered.append(row)\n",
    "    return filtered\n",
    "                \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.astype({'POS': str, 'CHR': str})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = populate_alternate(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "position_df = df[['CHR', 'POS']]\n",
    "alleles_df = df[['CHR', 'POS', 'Ancestralallele', 'Alternate']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>Ancestralallele</th>\n",
       "      <th>Alternate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>22</td>\n",
       "      <td>50704357</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>22</td>\n",
       "      <td>48228760</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>22</td>\n",
       "      <td>45549032</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>22</td>\n",
       "      <td>46771921</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>389</th>\n",
       "      <td>22</td>\n",
       "      <td>25914727</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>22</td>\n",
       "      <td>22547460</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>830</th>\n",
       "      <td>22</td>\n",
       "      <td>35534933</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>863</th>\n",
       "      <td>22</td>\n",
       "      <td>46500164</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>956</th>\n",
       "      <td>22</td>\n",
       "      <td>45279529</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1005</th>\n",
       "      <td>22</td>\n",
       "      <td>29854959</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHR       POS Ancestralallele Alternate\n",
       "6     22  50704357               C         T\n",
       "74    22  48228760               C         T\n",
       "215   22  45549032               T         C\n",
       "280   22  46771921               C         T\n",
       "389   22  25914727               G         C\n",
       "497   22  22547460               T         C\n",
       "830   22  35534933               G         C\n",
       "863   22  46500164               C         G\n",
       "956   22  45279529               A         G\n",
       "1005  22  29854959               T         C"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alleles_df[df['CHR']=='22']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "position_df.to_csv(open('/Users/estorrs/Documents/ding/ancestry/positions.tsv', 'w'), sep='\\t', header=None, index=None)\n",
    "alleles_df.to_csv(open('/Users/estorrs/Documents/ding/ancestry/alleles.tsv', 'w'), sep='\\t', header=None, index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
